virtool/analyses/api.py

```diff
@@ -242,3 +245,37 @@ async def blast(req: aiohttp.web.Request) -> aiohttp.web.Response:
     }
 
     return json_response(blast_data, headers=headers, status=201)
+
+
+@routes.patch("/api/analyses/{analysis_id}", schema={
+    "results": {"type": "dict", 'required': True}
+})
+async def patch_analysis(req: aiohttp.web.Request):
+    """Sets the result for an analysis and marks it as ready."""
+    db: DB = req.app["db"]
+    analyses: Collection = db.analyses
+    analysis_id: str = req.match_info["analysis_id"]
+
+    analysis_document: Dict[str, Any] = await analyses.find_one(dict(_id=analysis_id))
+
+    if not analysis_document:
+        return not_found(f"There is no analysis with id {analysis_id}")
+
+    if "ready" in analysis_document and analysis_document["ready"]:
+        return conflict("There is already a result for this analysis.")
+
+    await analyses.update_one(dict(_id=analysis_id), {
+        "$set": {
+            "results": (await req.json())["results"],
```

Comment:
It is best practice to explicitly deconstruct the request input and use the values to create your database query. This code will allow setting arbitrary fields in the database.